home/export_content_pages.py

```diff
@@ -0,0 +1,372 @@
+import copy
+import csv
+from dataclasses import asdict, astuple, dataclass, fields
+from itertools import zip_longest
+from json import dumps
+from math import ceil
+
+from django.http import HttpResponse  # type: ignore
+from openpyxl.styles import Border, Color, Font, NamedStyle, PatternFill, Side
+from openpyxl.utils import get_column_letter
+from openpyxl.workbook import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from wagtail import blocks  # type: ignore
+from wagtail.models import Locale, Page  # type: ignore
+from wagtail.query import PageQuerySet  # type: ignore
+
+from home.models import (
+    ContentPage,
+    ContentPageIndex,
+    HomePage,
+    MessengerBlock,
+    VariationBlock,
+    ViberBlock,
+    WhatsappBlock,
+)
+
+HP_CTYPE = HomePage._meta.verbose_name
+CP_CTYPE = ContentPage._meta.verbose_name
+CPI_CTYPE = ContentPageIndex._meta.verbose_name
+
+
+MsgBlocks = tuple[WhatsappBlock | None, MessengerBlock | None, ViberBlock | None]
+
+
+@dataclass
+class ExportRow:
+    structure: str = ""
+    message: int = 0
+    page_id: int = 0
+    slug: str = ""
+    parent: str = ""
+    web_title: str = ""
+    web_subtitle: str = ""
+    web_body: str = ""
+    whatsapp_title: str = ""
+    whatsapp_body: str = ""
+    whatsapp_template_name: str = ""
+    whatsapp_template_category: str = ""
+    example_values: str = ""
+    variation_title: str = ""
+    variation_body: str = ""
+    messenger_title: str = ""
+    messenger_body: str = ""
+    viber_title: str = ""
+    viber_body: str = ""
+    translation_tag: str = ""
+    tags: str = ""
+    quick_replies: str = ""
+    triggers: str = ""
+    locale: str = ""
+    next_prompt: str = ""
+    buttons: str = ""
+    image_link: str = ""
+    doc_link: str = ""
+    media_link: str = ""
+    related_pages: str = ""
+
+    @classmethod
+    def headings(cls) -> list[str]:
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str | int]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str | int, ...]:
+        return astuple(self)
+
+    def new_message_row(self, **fields: dict[str, str]) -> "ExportRow":
```

Comment:
Why do we take in a `fields` parameter here, but never use it?

Reply:
Leftover from a previous iteration of the code. I'll remove it, thanks.